Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -89,47 +89,30 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         }
     }
 
-    /// Holds the date when the status was last changed so we can send it out as additional information
-    /// in our status-change notifications.
-    ///
-    private var lastStatusChangeDate = Date()
-
-    private var connectionStatus: ConnectionStatus = .disconnected {
+    public var connectionStatus: ConnectionStatus = .disconnected {
         didSet {
-            if oldValue != connectionStatus {
-                lastStatusChangeDate = Date()
-                broadcastConnectionStatus()
+            guard connectionStatus != oldValue else {
+                return
             }
+
+            connectionStatusPublisher.send(connectionStatus)
```

Comment:
Slightly different structure, we now just publish to any interested subscriber that the value was changed instead of broadcasting distributed notifications (which aren't relevant for iOS).